src/server/game/Spells/Spell.cpp

```diff
@@ -5403,12 +5403,12 @@ void Spell::TakeAmmo()
             else
             {
                 // decrease items amount for stackable throw weapon
-                uint32 count = 1;
+                uint32 count = (sWorld->getBoolConfig(CONFIG_ENABLE_INFINITEAMMO) ? 0 : 1);
                 m_caster->ToPlayer()->DestroyItemCount(pItem, count, true);
             }
         }
         else if (uint32 ammo = m_caster->ToPlayer()->GetUInt32Value(PLAYER_AMMO_ID))
-            m_caster->ToPlayer()->DestroyItemCount(ammo, 1, true);
+            m_caster->ToPlayer()->DestroyItemCount(ammo, sWorld->getBoolConfig(CONFIG_ENABLE_INFINITEAMMO) ? 0 : 1, true);
```

Comment:
```suggestion
        else if (!sWorld->getBoolConfig(CONFIG_ENABLE_INFINITEAMMO))
            if (uint32 ammo = m_caster->ToPlayer()->GetUInt32Value(PLAYER_AMMO_ID))
                m_caster->ToPlayer()->DestroyItemCount(ammo, 1, true);
```

Reply:
Cant apply these automatically due to " Applying suggestions on deleted lines is currently not supported. " so i'll do a commit. Thanks for the improvement!